src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4706,34 +4706,17 @@ public T[] Items
         {
             get
             {
-                List<T> tempList = new List<T>();
-                IEnumerator<T> currentEnumerator = _enumerable.GetEnumerator();
-
-                if (currentEnumerator != null)
-                {
-                    for (_count = 0; currentEnumerator.MoveNext(); _count++)
-                    {
-                        tempList.Add(currentEnumerator.Current);
-                    }
-                }
-                if (_count == 0)
+                T[] array = _enumerable.ToArray();
+                if (array.Length == 0)
                 {
                     throw new SystemCore_EnumerableDebugViewEmptyException();
                 }
-
-                _cachedCollection = tempList.ToArray();
-                return _cachedCollection;
+                return array;
             }
         }
 
         [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
         private IEnumerable<T> _enumerable;
-
-        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
-        private T[] _cachedCollection;
```

Comment:
I experimented in that way when I first noticed the lack of `Dispose()` call and gave this a try, including a few obviously wrong things to check it was definitely using my code rather than something from a reference elsewhere or something. It certainly seems to be okay to that test.